OmiseSDK/Omise3DSViewController.swift

```diff
@@ -0,0 +1,131 @@
+        let viewController = storyboard.instantiateViewController(withIdentifier: "Default3DSVerificationController") as! Omise3DSViewController
+        viewController.authorizedURL = authorizedURL
+        viewController.delegate = delegate
+        
+        return viewController
+    }
+    
+    public override func loadView() {
+        super.loadView()
+        
+        view.addSubview(webView)
+        webView.navigationDelegate = self
+        
+        if #available(iOS 9.0, *) {
+            webView.topAnchor.constraint(equalTo: topLayoutGuide.bottomAnchor).isActive = true
+            webView.bottomAnchor.constraint(equalTo: bottomLayoutGuide.topAnchor).isActive = true
+            webView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
+            webView.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
+        } else {
+            let views = ["webView": webView] as [String: UIView]
+            let constraints = NSLayoutConstraint.constraints(withVisualFormat: "|[webView]|", options: [], metrics: nil, views: views) +
+                NSLayoutConstraint.constraints(withVisualFormat: "V:|[webView]|", options: [], metrics: nil, views: views)
+            view.addConstraints(constraints)
+        }
+    }
+    
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        start3DSProcess()
+    }
+    
+    private func start3DSProcess() {
+        guard let authorizedURL = authorizedURL else {
+            return
+        }
+        
+        let request = URLRequest(url: authorizedURL, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData, timeoutInterval: 60.0)
+        webView.load(request)
+    }
+    
+    fileprivate func verifyPaymentURL(_ url: URL) -> Bool {
+        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: true), let host = components.host else { return false }
+        
+        let pathComponents = components.path.characters.split(separator: "/").map(String.init)
+        return host.hasSuffix("omise.co") && host.hasPrefix("api") && pathComponents.first == "payments" && pathComponents.last == "complete"
+    }
+    
+    @IBAction func cancel3DSProcess(_ sender: UIBarButtonItem) {
+        delegate?.omise3DSViewControllerDidCancel(self)
+    }
+}
+
+extension Omise3DSViewController: WKNavigationDelegate {
+    public func webView(_ webView: WKWebView, decidePolicyFor navigationResponse: WKNavigationResponse, decisionHandler: @escaping (WKNavigationResponsePolicy) -> Swift.Void) {
+        readyToReturn = navigationResponse.response.url.map(verifyPaymentURL) ?? false
+        decisionHandler(.allow)
+    }
+    
+    public func webView(_ webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
+        if readyToReturn {
+            delegate?.omise3DSViewController(self, didFinish3DSProcessWithRedirectedURL: webView.url)
```

Comment:
Should we force-stop network communication from `WebView` here also? Just in case :p